### PR TITLE
Detect if add() is called while complete() is running

### PR DESCRIPTION
**Pull request by idodeclare**
This will detect the bizarre situation where `add()` is called while `complete()` is running, per my wild theory re #3210.

### CODE REVIEW

File: opengrok-indexer/src/main/java/org/opengrok/indexer/index/PendingFileCompleter.java [hunk 1]
```diff
@@ -171,6 +185,15 @@ public boolean add(PendingFileRenaming e) {
...
     public int complete() throws IOException {
+        completing = true;
+        try {
+            return complete2();
+        } finally {
+            completing = false;
+        }
+    }
+
+    private int complete2() throws IOException {
```
Comment (vladak) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/index/PendingFileCompleter.java:196]: use some more descriptive name ?

Comment (idodeclare) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/index/PendingFileCompleter.java:196]: I used to name these like `completeInner()` or `completeWork()` but I now like to keep the name even closer to the public method, and the single `2` character is the smallest difference, yet still communicates sequence. Thoughts?

Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/index/PendingFileCompleter.java:196]: perhaps call the outer method `completeWrapper()` ?

Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/index/PendingFileCompleter.java:196]: actually, `complete()` and `completeInner()` would be fine to me as well.

Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/index/PendingFileCompleter.java:196]: Also, I saw somewhere naming scheme such as `complete()` and `doComplete()`.

Comment (idodeclare) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/index/PendingFileCompleter.java]: done as `completeInner()`

**Comment by vladak** [comment]
Thanks !

**Pull request merged**